### DISCUSSION TITLE
required field validation is not working when schema defined in object type field.

### DISCUSSION THREAD

**Discussion by ATaiIsHere** [sensedeep/dynamodb-onetable#424] [General]
schema:
```
{
  version: "0.0.1",
  indexes: {
    primary: { hash: "pk", sort: "sk" },
  },
  models: {
    USER: {
      pk: { type: String, value: "USER#${id}" },
      sk: { type: String, value: "USER#${id}" },
      id: { type: String, required: true },
      privacySetting: {
        type: Object,
        schema: {
          privacyType: { type: String, required: true },
          updated: { type: String, required: true },
        },
        required: true,
      },
    },
  },
}
```
---

when I update user by
```
model.update({
  id,
  privacySetting: {
    privacyType: null,
    updated: new Date().toISOString(),
  },
});
```
I got a error msg as  "Value not defined for required field".

---
use
```
model.update({
  id,
  privacySetting: {
    privacyType: undefine,
    updated: new Date().toISOString(),
  },
});
```
I can remove privacySetting.privacyType successfully,
but I think it was right to response me the same error.

**Comment by mobsense** [comment]
Not sure if your example has the error, but your schema uses "type" not "privacyType".

```
          type: { type: String, required: true },
```

Because you are not defining "type", you should see the error you are seeing.

Setting a property to null is handled differently depending on your params.nulls setting. If default (false), then a property set to null will remove the property.

**Reply by ATaiIsHere** [reply]
Sorry for my mistake, I have corrected my example.

**Reply by ATaiIsHere** [reply]
https://github.com/sensedeep/dynamodb-onetable/blob/v2.5.1/src/Model.js#L1514

the condition: `(op == 'update' && properties[field.name] === null)`
I think it should be`(op == 'update' && properties[field.name] == null)` when the field in a `Object` type field like field in `privacySetting`.


no matter
```
{
  privacyType: null,
  updated: '2022-11-30T11:00:00Z',
}
```
or
```
{
  privacyType: undefine,
  updated:  '2022-11-30T11:00:00Z',
}
```
it will all save as `{ "updated": "2022-11-30T11:00:00Z" }` when field type is 'Object'.
